test/testtildeexpansion.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QDir>
+#include <QObject>
+#include <QString>
+#include <QTest>
+
+#include "base/path.h"
+#include "base/utils/fs.h"
+
+class TestTildeExpansion : public QObject
```

Comment:
You should copy existing code and rename the fields. You are missing a few directives and doesn't follow the coding style.